lib/ansible/modules/cloud/azure/azure_rm_storageaccount_keys.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2016 Amanvir Mundra, <@amanvirmundra>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_storageaccount_keys
+
+version_added: "2.6"
+
+short_description: Retrieve Azure Storage Account Keys.
+
+description:
+    - Retrive keys for a Storage Account.
+
+options:
+    resource_group:
+        description:
+            - This is the name of the resource group for which you want to retrieve Keys
+        required: true
+        default: null
+    name:
+        description:
+            - Name of the Storage Account instance.
+        required: true
+        default: null
+        aliases:
+            - resource_group_name
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - Amanvir Mundra (@amanvirmundra)
+
+'''
+
+EXAMPLES = '''
+- name: Get Storage account keys
+  azure_rm_storageaccount_keys:
+    resource_group: resource_group
+    name: name
+'''
+
+RETURN = '''
+key1:
+    description: Key1 for azure storage account
+    returned: always
+    type: str
+key2:
+    description: Key2 for azure storage account
+    returned: always
+    type: str
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMStorageAccountKeys(AzureRMModuleBase):
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', aliases=['resource_group_name'], required=True),
+            name=dict(type='str', required=True)
+        )
+        self.results = dict(
+            changed=False,
+            key1=None,
+            key2=None
+        )
+        self.resource_group = None
+        self.name = None
+
+        super(AzureRMStorageAccountKeys, self).__init__(self.module_arg_spec, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        try:
+            account_keys = self.storage_client.storage_accounts.list_keys(self.resource_group, self.name)
+        except CloudError:
+            pass
+
+        self.results['Key1'] = account_keys.keys[0].value
+        self.results['Key2'] = account_keys.keys[1].value
```

Comment:
does storage account always have 2 keys?
i think it would be better to loop through account_keys instead of assigning two keys with prefixed names.

Reply:
Azure has a concept of Primary and Secondary keys, so yes, for now they have only two keys. But I agree with you, looping would be future safe. I was having some backward compatibility (python 2.6v) issues with using a looping syntax, so took the easy way out.
I'll fix that.